src/libraries/System.Net.HttpListener/src/System/Net/Managed/HttpConnection.cs

```diff
@@ -209,8 +212,8 @@ public HttpRequestStream GetRequestStream(bool chunked, long contentlength)
         {
             if (_requestStream == null)
             {
-                byte[] buffer = _memoryStream.GetBuffer();
-                int length = (int)_memoryStream.Length;
+                byte[] buffer = _memoryStream!.GetBuffer();
+                int length = (int)_memoryStream!.Length;
                 _memoryStream = null;
```

Comment:
@aik-jahoda shouldn't we call `_memoryStream.Dispose()` here? I find it weird that it is being set to `null`.

Reply:
I wouldn't change behaviour now but create follow up work for .NET 6. I'm fine with nullable _memoryStream since we have this discussion recorder here for future fix.